app/javascript/packs/body.jsx

```diff
@@ -0,0 +1,183 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+import ReactPaginate from 'react-paginate';
+import Timelogs from './timelogs.jsx'
+import NewTimelog from './new_timelog.jsx'
+
+class Body extends React.Component {
+
+  constructor(props) {
+    super(props);
```

Comment:
The constructor accepts props and context. You basically removed context from this component.

```
constructor(...args) {
  super(...args);
}
```

Here you are passing all the arguments to parent constructor. See spread operator ES6